docs/Host_Validation_Architecture.md

```diff
@@ -0,0 +1,25 @@
+# Host Validation and Testing
+
+__Date:__  7/6/2017
+
+__Objective:__ To determine the best strategy to validate hosts have joined the Kubernetes cluster successfully and are ready to schedule workloads
+
+## Problem Statement
+The Promenade project is a simple and efficient tool for orchestrating the deployment of self-hosted Kubernetes clusters.  However, Promenade currently lacks the ability to validate that clusters have been deployed successfully and are actually ready to schedule workloads.   I would like to propose a series of tests and validations that can be included with Promenade which will ensure the Kubernetes clusters are working as intended at the time of deployment, and can be executed at any point in time  after the cluster has been deployed to ensure it is continuing to function as intended for Day 2 operations. This tool should be able to identify potential problems with the Kubernetes cluster and provide the user with a comprehensive debugging output which highlights the identified issues.
+
+## Identified Tests
+I have identified the following tests which I believe to be a good base-line for basic Kubernetes smoke testing at the time of cluster deployment and during cluster life-cycle management:
+
+1. __Generate SHA256 hashes of all Promenade Generated configuration files and certificates:__ This process will run during the initial deployment of all the Kubernetes hosts in the cluster. Promenade will write a file to `/var/log/promenade/promenade_file_hash.log` with the path to the file as well as the SHA256 hash of that file when it was originally generated. Subsequent runs of the host validation script will re-calculate the SHA256 hashes of the files as they are presently, and compare them with the values written in the log file at the time of deployment. If the file hashes differ, the host validation script will log a failure and provide the user with a list of files that have potentially been modified.
```

Comment:
The purpose of this test would be to check to see if the files generated by Promenade have been modified in the time since the deployment. If the SHA256 hashes created at the time of deployment do not line up with the SHA256 hashes at the time the test is run, then a change has been made to that file which may (or may not) impact the functionality of the K8s cluster. The purpose of this test is to provide a list of files that have changed since deployment to help with future debugging.